service/controller/aws/resource_set.go

```diff
@@ -419,11 +420,11 @@ func newResourceSet(config resourceSetConfig) (*controller.ResourceSet, error) {
 			return false
 		}
 
-		if key.OperatorVersion(&cr) == VersionBundle().Version {
+		if key.OperatorVersion(&cr) == project.BundleVersion() {
```

Comment:
When this is `legacy` it should not be concerned with the version label I think. We only introduced this for CAPI.

Reply:
We should have it for legacy objects as well. We'll need that for label selector. Whatever is flat must  use it eventually.

Reply:
Why is that then only in the AWS resource set? Then I would expect this to also be true for Azure and KVM.

Reply:
You're right here.

Reply:
I made proper notes. We'll sort that later.